go-controller/pkg/cni/helper_linux.go

```diff
@@ -66,25 +66,29 @@ func setupInterface(netns ns.NetNS, hostIfaceName, ifName, macAddress, ipAddress
 		contIface.Mac = macAddress
 		contIface.Sandbox = netns.Path()
 
-		addr, err := netlink.ParseAddr(ipAddress)
-		if err != nil {
-			return err
-		}
-		err = netlink.AddrAdd(link, addr)
-		if err != nil {
-			return fmt.Errorf("failed to add IP addr %s to %s: %v", ipAddress, contIface.Name, err)
-		}
-
-		gw := net.ParseIP(gatewayIP)
-		if gw == nil && isDefaultInterface {
-			return fmt.Errorf("parse ip of gateway failed")
-		}
-		if gw != nil {
-			err = ip.AddRoute(nil, gw, link)
-			if err != nil {
-				return err
-			}
-		}
+        if ipAddress != "" {
+		    addr, err := netlink.ParseAddr(ipAddress)
+		    if err != nil {
+			    return err
+		    }
+		    err = netlink.AddrAdd(link, addr)
+		    if err != nil {
+			    return fmt.Errorf("failed to add IP addr %s to %s: %v", ipAddress, contIface.Name, err)
+		    }
+        }
+
+        if gatewayIP != "" {
+    		gw := net.ParseIP(gatewayIP)
+	    	if gw == nil && isDefaultInterface {
```

Comment:
Now that you've added 'if gatewayIP == "" ', it seems that the "isDefaultInterface" is redundant.

Reply:
so, if gw IP exists and is malformed, we want to submit an error for any interface?

Reply:
yes